### PR TITLE
Add initial MacOS support

### PR DESCRIPTION
**Pull request by RickyGrassmuck**
This commit adds the ability to build and run the application on MacOS.

This will have to be built and run by the users at this time as no work has been done to make a pre-built package yet.

I've not created a proper Mac OS Application package for distribution before but I do know that it requires obtaining an apple developer account in order to sign the package. I may do this once the package has gotten a little more refined and is actually ready for that.

That said, there's a good bit of work that would need to be done to make this more usable (IE. loading the `GeneratorConfig/DeviceSources` from a different location since Mac Apps aren't distributed as a single executable).

### CODE REVIEW

**Comment by Mysticial** [comment]
Oh nice!

File: Repository/DeviceSource/ShinyHunt-Regi.c [hunk 1]
```diff
@@ -1,22 +1,4 @@
-/*  Pokemon Sword & Shield: Soft-Reset for Regis
```
Comment (Mysticial) [inline Repository/DeviceSource/ShinyHunt-Regi.c:1]: Let's start by reverting the two files that the UI overwrote. This one.

**Comment by RickyGrassmuck** [comment]
Whoops, thought I kept those outta the commit. I'll get those changes pushed up

**Comment by RickyGrassmuck** [comment]
Git was not cooperating with removing those changes so I opened #5 with the requested changes. Closing this PR out.

**Pull request closed without merging**